Examples/Run/Geant4/HepMC/EventRecordingExample.cpp

```diff
@@ -0,0 +1,82 @@
+#include "ActsExamples/DD4hepDetector/DD4hepGeometryService.hpp"
+#include "ActsExamples/EventRecording/EventRecording.hpp"
+#include "ActsExamples/Framework/Sequencer.hpp"
+#include "ActsExamples/Geant4/Geant4Options.hpp"
+#include "ActsExamples/Geant4DD4hep/DD4hepDetectorConstruction.hpp"
+#include "ActsExamples/Geometry/CommonGeometry.hpp"
+#include "ActsExamples/Io/Csv/CsvOptionsReader.hpp"
+#include "ActsExamples/Io/Csv/CsvParticleReader.hpp"
+#include "ActsExamples/Io/HepMC3/HepMC3Options.hpp"
+#include "ActsExamples/Io/HepMC3/HepMC3Writer.hpp"
+#include "ActsExamples/Options/CommonOptions.hpp"
+#include "ActsExamples/Utilities/Paths.hpp"
+#include <fstream>
+#include <string>
+#include <boost/program_options.hpp>
+
+int main(int argc, char* argv[]) {
+  // Declare the supported program options.
+  // Setup and parse options
+  auto desc = ActsExamples::Options::makeDefaultOptions();
+  ActsExamples::Options::addSequencerOptions(desc);
+  ActsExamples::Options::addOutputOptions(desc);
+  ActsExamples::Options::addDD4hepOptions(desc);
+  ActsExamples::Options::addGeant4Options(desc);
+  ActsExamples::Options::addHepMC3WriterOptions(desc);
+
+  auto vm = ActsExamples::Options::parse(desc, argc, argv);
+  if (vm.empty()) {
+    return EXIT_FAILURE;
+  }
+
+  auto logLevel = ActsExamples::Options::readLogLevel(vm);
+
+  ActsExamples::Sequencer sequencer(
+      ActsExamples::Options::readSequencerConfig(vm));
+
+  // Read particles (initial states) and clusters from CSV files
+  auto particleReader = ActsExamples::Options::readCsvParticleReaderConfig(vm);
+  particleReader.inputStem = "particles";
+  particleReader.outputParticles = "particles";
+  sequencer.addReader(std::make_shared<ActsExamples::CsvParticleReader>(
+      particleReader, logLevel));
+
+  // Prepare the detector
+  auto dd4hepCfg = ActsExamples::Options::readDD4hepConfig(vm);
+  auto geometrySvc =
+      std::make_shared<ActsExamples::DD4hep::DD4hepGeometryService>(dd4hepCfg);
+  std::unique_ptr<G4VUserDetectorConstruction> g4detector =
+      std::make_unique<ActsExamples::DD4hepDetectorConstruction>(
+          *geometrySvc->lcdd());
+
+  // Prepare the recording
+  ActsExamples::EventRecording::Config erConfig;
+  erConfig.inputParticles = particleReader.outputParticles;
+  erConfig.outputHepMcTracks = "geant-event";
+  erConfig.detectorConstruction = std::move(g4detector);
+  erConfig.seed1 = vm["g4-rnd-seed1"].as<unsigned int>();
+  erConfig.seed2 = vm["g4-rnd-seed2"].as<unsigned int>();
+
+  // Create the writer
+  auto hepMC3WriterConfig = ActsExamples::Options::readHepMC3WriterOptions(vm);
+  hepMC3WriterConfig.inputEvents = erConfig.outputHepMcTracks;
+
+  // Add to the sequencer
+  sequencer.addAlgorithm(std::make_shared<ActsExamples::EventRecording>(
+      std::move(erConfig), logLevel));
+  sequencer.addWriter(std::make_shared<ActsExamples::HepMC3AsciiWriter>(
+      std::move(hepMC3WriterConfig), logLevel));
+
+  // Run
+  sequencer.run();
```

Comment:
```suggestion
  return sequencer.run();
```
See `Sequencer` docs for details.

Reply:
done